examples/alerts/alerts.yaml

```diff
@@ -328,7 +318,10 @@ groups:
       runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanossidecarunhealthy
       summary: Thanos Sidecar is unhealthy.
     expr: |
-      time() - max by (job, instance) (thanos_sidecar_last_heartbeat_success_time_seconds{job=~".*thanos-sidecar.*"}) >= 240
+      time() - max by (pod, job, instance) (thanos_sidecar_last_heartbeat_success_time_seconds{job=~".*thanos-sidecar.*"}) >= 240
+      AND on (pod) (
+        min by (pod) (prometheus_tsdb_data_replay_duration_seconds) != 0
```

Comment:
This is quite neat actually! We know the duration is done after finish so we expect Prometheus to be healthy.

But instead of doing this long term, can we see if Prometheus expose any health metric and use that instead? 🤔 

Anyway, for now this will do 👍🏽

Reply:
AFAIK, I couldn't find a reasonable one. May be @dgrisonnet @paulfantom know something?

Reply:
There is none, but we can add one if we want. Question, do we want (:

Reply:
I'm not really sure. As you mentioned, I will go ahead with the current solution and improve it in future.